packages/main/src/Slider.js

```diff
@@ -0,0 +1,271 @@
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @private
+		 */
+		_expanded: {
+			type: Boolean,
+		},
+
+		_items: {
+			type: Object,
+		},
+	},
+	slots: /** @lends sap.ui.webcomponents.main.Slider.prototype */ {
+
+	},
+	events: /** @lends sap.ui.webcomponents.main.Slider.prototype */ {
+		/**
+		 * Fires when the slider is expanded.
+		 */
+		expand: {},
+
+		/**
+		 * Fires when the slider is collapsed.
+		 */
+		collapse: {},
+
+		/**
+		 *  Fires when new value is selected.
+		 */
+		valueSelect: {
+			value: {
+				type: String,
+			},
+		},
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ *
+ * <h3>Usage</h3>
+ *
+ * For the <code>ui5-slider</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/Slider.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.Slider
+ * @extends UI5Element
+ * @tagname ui5-slider
+ * @public
+ */
+class Slider extends UI5Element {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return SliderCss;
+	}
+
+	static get template() {
+		return SliderTemplate;
+	}
+
+	constructor() {
+		super();
+		this._currentElementIndex = 0;
+		this._itemCellHeight = 0;
+	}
+
+	onBeforeRendering() {
+		this._updateItemCellHeight();
+	}
+
+	_updateItemCellHeight() {
+		this._itemCellHeight = this.shadowRoot.querySelectorAll(".ui5-slider-item").length && Number(getComputedStyle(this.shadowRoot.querySelector(".ui5-slider-item")).getPropertyValue("--_ui5_slider_item_height").replace("rem", ""));
+	}
+
+	onAfterRendering() {
+		this.shadowRoot.querySelector(".ui5-slider-wrapper").addEventListener("wheel", e => {
+			this._handleWheel(e);
+		});
+
+		if (this._expanded) {
+			const elements = this.shadowRoot.querySelectorAll(".ui5-slider-item");
+			for (let i = 0; i < elements.length; i++) {
+				if (elements[i].textContent === this.value) {
+					this._selectElement(elements[i]);
+					return true;
+				}
+			}
+
+			this._selectElement(elements[0]);
+		}
+	}
+
+	get items() {
+		return this._items;
+	}
+
+	_handleWheel(e) {
+		if (!e){
+			return;
+		}
+
+		e.stopPropagation();
+		e.preventDefault();
+
+		if (e.timeStamp === this._prevWheelTimestamp || !this._expanded) {
+			return;
+		}
+
+		if (e.deltaY > 0) {
+			this._onArrowUp(e);
+		} else if (e.deltaY < 0) {
+			this._onArrowDown(e);
+		}
+
+		this._prevWheelTimestamp = e.timeStamp;
+	}
+
+	_onclick(e) {
+		if (!e.target.classList.contains("ui5-slider-item")) {
+			return;
+		}
+
+		if (this._expanded) {
+			this.value = e.target.textContent;
+			this._selectElement(e.target);
+		} else {
+			this._expanded = true;
+		}
+	}
+
+	expandSlider() {
+		this._expanded = true;
+		this.fireEvent("expand", {});
+	}
+
+	collapseSlider() {
+		this._expanded = false;
+		this.fireEvent("collapse", {});
+	}
+
+	_selectElement(element) {
+		if (element && this._items.indexOf(element.textContent) > -1) {
```

Comment:
this._items.indexOf throws an exception if no items, this is the case in Slider test page. it is used at several places and potentially they can throw as well. You  can use the

get items() {
		return this._items;
	}

modified

get items() {
		return this._items || [];
}

Reply:
Done